addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/handler/OpenDaikinAcUnitHandler.java

```diff
@@ -0,0 +1,246 @@
+        }
+    }
+
+    private void pollStatus() throws IOException {
+        ControlInfo controlInfo = webTargets.getControlInfo();
+        updateStatus(ThingStatus.ONLINE);
+        if (controlInfo != null) {
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
+            updateTempState(OpenDaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp);
+
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_MODE, new StringType(controlInfo.mode.name()));
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_FAN_SPEED, new StringType(controlInfo.fanSpeed.name()));
+        }
+
+        SensorInfo sensorInfo = webTargets.getSensorInfo();
+        if (sensorInfo != null) {
+            if (sensorInfo.indoortemp.isPresent()) {
+                updateTempState(OpenDaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.indoortemp.get());
+            } else {
+                updateState(OpenDaikinBindingConstants.CHANNEL_INDOOR_TEMP, UnDefType.UNDEF);
+            }
+
+            if (sensorInfo.outdoortemp.isPresent()) {
+                updateTempState(OpenDaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, sensorInfo.outdoortemp.get());
+            } else {
+                updateState(OpenDaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, UnDefType.UNDEF);
+            }
+
+            if (sensorInfo.indoorhumidity.isPresent()) {
+                updateState(OpenDaikinBindingConstants.CHANNEL_HUMIDITY,
+                        new DecimalType(sensorInfo.indoorhumidity.get()));
+            } else {
+                updateState(OpenDaikinBindingConstants.CHANNEL_HUMIDITY, UnDefType.UNDEF);
+            }
+        }
+    }
+
+    private void updateTempState(String channel, double temp) {
+        DecimalType tempToUpdateWith;
+        if (useFahrenheitForChannel(channel)) {
+            tempToUpdateWith = new DecimalType(cToF(temp));
+        } else {
+            tempToUpdateWith = new DecimalType(temp);
+        }
+
+        updateState(channel, tempToUpdateWith);
+    }
+
+    private void changePower(boolean power) throws OpenDaikinCommunicationException {
+        ControlInfo info = webTargets.getControlInfo();
+        info.power = power;
+        webTargets.setControlInfo(info);
+    }
+
+    private void changeSetPoint(double temp) throws OpenDaikinCommunicationException {
+        ControlInfo info = webTargets.getControlInfo();
+
+        if (useFahrenheitForChannel(OpenDaikinBindingConstants.CHANNEL_AC_TEMP)) {
+            info.temp = fToC(temp);
+        } else {
+            info.temp = temp;
+        }
+
+        webTargets.setControlInfo(info);
+    }
+
+    private boolean useFahrenheitForChannel(String channel) {
+        return this.getThing().getChannel(channel) != null
+                && this.getThing().getChannel(channel).getConfiguration().containsKey("useFahrenheit")
```

Comment:
Put `useFahrenheit` in `OpenDaikinBindingConstants` and compare `Boolean.TRUE.equals(getThing().getChannel(channel).getConfiguration().get("useFahrenheit"))` then no contains check is needed.